compiler/rustc_const_eval/src/interpret/eval_context.rs

```diff
@@ -613,10 +613,9 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                             if sized_size == Size::ZERO {
                                 return Ok(None);
                             } else {
-                                span_bug!(
-                                    self.cur_span(),
+                                throw_ub!(InvalidMeta(
                                     "Fields cannot be extern types, unless they are at offset 0"
-                                )
+                                ))
                             }
```

Comment:
What is this about?

If the assertion "Fields cannot be extern types, unless they are at offset 0" is wrong, then we have to fix Miri to support them. Turning this into `throw_ub!` is likely not correct.

Reply:
I expect so. For now, this is a dirty fix to make const-prop skip processing an assignment and pass bootstrap. `throw_ub` was closest to hand.

This assertion was triggered by const-prop trying to downcast an `Option<&ty::List<ty::Predicate<'_>>>` as `Some` when the value had been set to `None`. `ty::List` uses an extern type as last field to be unsized.
What would be the correct fix?

Reply:
Dirty tricks should at least have `// FIXME HACK` to warn the future reader. ;)

I don't know how other parts of the compiler handle size and alignment of types with an `extern type` unsized tail. Miri should be consistent with whatever the rest of the compiler does. :)

Reply:
If the unsized part is an extern type then code generation uses static size and alignment.

Reply:
The issue with extern types should be fixed by #91910.